optuna/pruners/hyperband.py

```diff
@@ -70,20 +78,42 @@ class HyperbandPruner(BasePruner):
     """
 
     def __init__(
-        self, min_resource=1, reduction_factor=3, n_brackets=4, min_early_stopping_rate_low=0
-    ):
-        # type: (int, int, int, int) -> None
-
+        self,
+        max_resource: int = 100,
+        min_resource: int = 1,
+        reduction_factor: int = 3,
+        n_brackets: Optional[int] = None,
+        min_early_stopping_rate_low: int = 0,
+    ) -> None:
+        self._max_resource = max_resource
```

Comment:
Is/Will `max_resource` be used for any other purpose than computing the number of brackets? If not, we don't have to store it as a (private) class member.

Reply:
In the current implementation, we do not use `max_resource` except for computing the number of brackets. I will remove it. (But, it will be used for adaptive `min_resource` allocation as I'm planning now.)

Reply:
I think `max_resource` is also used in the "adaptive `min_resource` allocation", but there is no need to store it.